t/proxyunits.t

```diff
@@ -444,14 +504,36 @@ check_version($ps);
     print $be "END\r\n";
     is(scalar <$ps>, "END\r\n", "ltrimkey END");
 
-    # token(n) fetch
-    # token(n, "replacement")
-    # token(n, "") removal
-    # ntokens()
-    # command() integer
+    # return the number of token as the value and verify.
+    print $ps "mg /ntokens/test c v\r\n";
+    is(scalar <$ps>, "VA 1 C123 v\r\n", "request:key()");
+    is(scalar <$ps>, "4\r\n", "request:ntokens() value");
+
+    # replace a token and verify request received by backend
+    print $ps "ms /token/replacement 2 C123\r\nhi\r\n";
+    is(scalar <$be>, "ms /token/replacement 2 C456\r\n", "token replacement");
+    is(scalar <$be>, "hi\r\n", "token replacement");
+    print $be "NF\r\n";
+    is(scalar <$ps>, "NF\r\n", "NF");
+
+    # remove a token and verify request received by backend.
+    print $ps "ms /token/removal 2 C123\r\nhi\r\n";
+    is(scalar <$be>, "ms /token/removal 2 \r\n", "token removal, note the space");
+    is(scalar <$be>, "hi\r\n", "token removal");
+    print $be "NF\r\n";
+    is(scalar <$ps>, "NF\r\n", "NF");
+
+    # fetch a token and verify in lua then return HD if matched.
+    print $ps "ms /token/fetch 2\r\nhi\r\n";
+    is(scalar <$ps>, "HD\r\n", "HD is received");
+
+    # TODO: command() integer
     #
-    # meta:
-    # has_flag("F")
+
+    # check flag and return HD if found.
```

Comment:
for completeness; run a couple tests with the checked flag in different positions and +/- a token.

Reply:
ack